### DISCUSSION TITLE
Default defineOgImageComponent for all pages

### DISCUSSION THREAD

**Discussion by AdamBD** [harlan-zw/nuxt-seo#232] [Q&A]
Hello,

Is there a way to define a default og:image for all pages?

**Comment by harlan-zw** [comment]
Hi, yes you can either add it to `app.vue` or use [route rules](https://nuxtseo.com/og-image/guides/route-rules).

Something like this should work.

```ts
export default defineNuxtConfig({
  routeRules: {
    '/**': {
      ogImage: {
        component: 'NuxtSeo',
        props: { icon: 'carbon:image-search' }
      }
    }
  }
})
```

**Reply by AdamBD** [reply]
thank you!

**Reply by AdamBD** [reply]
@harlan-zw I am trying to create a template from scratch now.

Following the documentation, I created 

```
	routeRules: { 
		"/**": {
			ogImage: {
				component: "~/components/OgImage/BlogPost",
		//		props: { title: "cool stuff", icon: "fa6-regular:hand-peace" },
			},
		},
	},
```

however when running the build I get this error `Error: [GET] "/__og-image__/image/og.json": 404 Island component not found: ~/components/OgImage/BlogPost`

I have also tried just `component: "BlogPost"`, which leads to the same error

**Comment by FabianMontoya** [comment]
Im having the same error, is very frustrated, only works with the rebuild templates, but if I want use a jpeg image or a custom component, I can't :(
